File: backend/internal/models/relationship.go
// relationship.go defines the Relationship model representing a directed edge
// between two Members (e.g. parent-child, spouse). It stores relationship type
// and optional metadata such as marriage date.
package models

import "time"

type Relationship struct {
	ID               string    `db:"id" json:"id"`
	ClanID           string    `db:"clan_id" json:"clan_id"`
	FromUserID       string    `db:"from_user_id" json:"from_user_id"`
	ToMemberID       string    `db:"to_member_id" json:"to_member_id"`
	RelationshipType string    `db:"relationship_type" json:"relationship_type"`
	IsInferred       bool      `db:"is_inferred" json:"is_inferred"`
	Status           string    `db:"status" json:"status"`
	SubmittedBy      *string   `db:"submitted_by" json:"submitted_by,omitempty"`
	CreatedAt        time.Time `db:"created_at" json:"created_at"`
	UpdatedAt        time.Time `db:"updated_at" json:"updated_at"`
}

File: frontend/src/types/relationship.ts
export type RelationshipType =
  | 'child' | 'parent' | 'spouse' | 'sibling'
  | 'uncle' | 'aunt' | 'cousin' | 'second_cousin'
  | 'grandparent' | 'grandchild' | 'nephew' | 'niece'
  | 'in_law' | 'step_parent' | 'step_child' | 'half_sibling'
  | 'co_wife'

export interface Relationship {
  id: string
  clan_id: string
  from_user_id: string
  to_member_id: string
  relationship_type: RelationshipType
  is_inferred: boolean
  status: 'active' | 'pending' | 'conflicted'
  submitted_by?: string
  created_at: string
  updated_at: string
}

export interface TreeData {
  members: import('./member').Member[]
  relationships: Relationship[]
}
